--- a/xbmc/windows/GUIWindowSystemInfo.cpp	2016-01-18 10:47:40.946673609 +0400
+++ b/xbmc/windows/GUIWindowSystemInfo.cpp	2016-01-18 11:22:57.376833638 +0400
@@ -57,7 +57,7 @@
     case GUI_MSG_WINDOW_INIT:
     {
       CGUIWindow::OnMessage(message);
-      SET_CONTROL_LABEL(52, CSysInfo::GetAppName() + " " + CSysInfo::GetVersion());
+      SET_CONTROL_LABEL(52, CSysInfo::GetAppName() + " " + CSysInfo::GetVersion() + " - built for AppleTV");
       SET_CONTROL_LABEL(53, CSysInfo::GetBuildDate());
       CONTROL_ENABLE_ON_CONDITION(CONTROL_BT_PVR, PVR::CPVRManager::GetInstance().IsStarted());
       return true;

