--- a/driver/linux/crystalhd_linkfuncs.c
+++ b/driver/linux/crystalhd_linkfuncs.c
@@ -553,16 +553,11 @@ uint32_t link_GetPicInfoLineNum(struct crystalhd_dio_req *dio, uint8_t *base)
 {
 	uint32_t PicInfoLineNum = 0;
 
-	if (dio->uinfo.b422mode == MODE422_YUY2) {
+	if (dio->uinfo.b422mode) {
 		PicInfoLineNum = ((uint32_t)(*(base + 6)) & 0xff)
 			| (((uint32_t)(*(base + 4)) << 8)  & 0x0000ff00)
 			| (((uint32_t)(*(base + 2)) << 16) & 0x00ff0000)
 			| (((uint32_t)(*(base + 0)) << 24) & 0xff000000);
-	} else if (dio->uinfo.b422mode == MODE422_UYVY) {
-		PicInfoLineNum = ((uint32_t)(*(base + 7)) & 0xff)
-			| (((uint32_t)(*(base + 5)) << 8)  & 0x0000ff00)
-			| (((uint32_t)(*(base + 3)) << 16) & 0x00ff0000)
-			| (((uint32_t)(*(base + 1)) << 24) & 0xff000000);
 	} else {
 		PicInfoLineNum = ((uint32_t)(*(base + 3)) & 0xff)
 			| (((uint32_t)(*(base + 2)) << 8)  & 0x0000ff00)
@@ -588,14 +583,10 @@ uint32_t link_GetMode422Data(struct crystalhd_dio_req *dio,
 	else
 		offset = 0;
 
-	if (dio->uinfo.b422mode == MODE422_YUY2) {
+	if (dio->uinfo.b422mode) {
 		for (i = 0; i < 4; i++)
 			((uint8_t*)tmp)[i] =
 				((uint8_t*)pPicInfoLine)[(offset + i) * 2];
-	} else if (dio->uinfo.b422mode == MODE422_UYVY) {
-		for (i = 0; i < 4; i++)
-			((uint8_t*)tmp)[i] =
-				((uint8_t*)pPicInfoLine)[(offset + i) * 2 + 1];
 	}
 
 	return val;
@@ -742,12 +733,9 @@ bool link_GetPictureInfo(struct crystalhd_hw *hw, uint32_t picHeight, uint32_t p
 	if (res != 0)
 		goto getpictureinfo_err;
 
-	if (dio->uinfo.b422mode == MODE422_YUY2) {
+	if (dio->uinfo.b422mode) {
 		for (i = 0; i < 4; i++)
 			((uint8_t *)tmp)[i] = ((uint8_t *)dio->pib_va)[i * 2];
-	} else if (dio->uinfo.b422mode == MODE422_UYVY) {
-		for (i = 0; i < 4; i++)
-			((uint8_t *)tmp)[i] = ((uint8_t *)dio->pib_va)[(i * 2) + 1];
 	} else
 		pic_number = *(uint32_t *)(dio->pib_va);
 

